gossip/client.go

```diff
@@ -84,9 +84,6 @@ func (c *client) start(g *Gossip, done chan *client, context *rpc.Context, stopp
 		// Start gossipping and wait for disconnect or error.
 		c.lastFresh = time.Now().UnixNano()
 		c.err = c.gossip(g, stopper)
-		if c.err != nil {
-			c.rpcClient.Close()
```

Comment:
I think we still need to close the client if we're bootstrapping (and `context.DisableCache` is true).